src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Net;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    [Export(typeof(ISonarLintDaemon))]
+    class SonarLintDaemon : ISonarLintDaemon
+    {
+        public const string daemonVersion = "2.12.0.492";
+        private const string uriFormat = "https://repox.sonarsource.com/sonarsource-public-builds/org/sonarsource/sonarlint/core/sonarlint-daemon/{0}/sonarlint-daemon-{0}-windows.zip";
+        private readonly string version;
+        private readonly string tmpPath;
+        private readonly string storagePath;
+        private int port;
+
+        private Process process;
+
+        public SonarLintDaemon() : this(daemonVersion, Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), Path.GetTempPath())
+        {
+        }
+
+        public SonarLintDaemon(string version, string storagePath, string tmpPath)
+        {
+            this.version = version;
+            this.tmpPath = tmpPath;
+            this.storagePath = storagePath;
+        }
+
+        public void Dispose()
+        {
+            if (IsRunning())
+            {
+                Stop();
+            }
+        }
+
+        public void Install()
+        {
+            Download();
+            Unzip();
+        }
+
+        public bool IsRunning()
```

Comment:
I would make this a property but it is not a big deal:

    public bool IsRunnint => process != null && !process.HasExited;